src/CMakeLists.txt

```diff
@@ -94,53 +94,65 @@ add_custom_command(
     # In case the DEPFILE possibility is not supported the complete list of lex include files for the dependency has to be used
     set(LEX_INC_FILES)
 endif()
 
-# unfortunately ${LEX_FILES_H} and ${LEX_FILES_CPP} don't work in older versions of CMake (like 3.6.2) for add_library
+set(LEX_FILES_H)
+set(LEX_FILES_CPP)
+set(_depfile_args)
+
 foreach(lex_file ${LEX_FILES})
-    set(LEX_FILES_H ${LEX_FILES_H} " " ${GENERATED_SRC}/${lex_file}.l.h CACHE INTERNAL "Stores generated files")
-    set(LEX_FILES_CPP ${LEX_FILES_CPP} " " ${GENERATED_SRC}/${lex_file}.cpp CACHE INTERNAL "Stores generated files")
+    # configimpl is handled specially (it's in the static lib)
+    if (NOT lex_file STREQUAL "configimpl")
+        list(APPEND LEX_FILES_H ${GENERATED_SRC}/${lex_file}.l.h)
+        list(APPEND LEX_FILES_CPP ${GENERATED_SRC}/${lex_file}.cpp)
+    endif()
+
+    set(_generated_files
+        ${GENERATED_SRC}/${lex_file}.l
+        ${GENERATED_SRC}/${lex_file}.corr
+        ${GENERATED_SRC}/${lex_file}.d)
 
     if (depfile_supported)
-        add_custom_command(
-            COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_LIST_DIR}/pre_lex.py ${CMAKE_CURRENT_LIST_DIR}/${lex_file}.l  ${GENERATED_SRC}/${lex_file}.l ${GENERATED_SRC}/${lex_file}.corr ${GENERATED_SRC}/${lex_file}.d ${CMAKE_CURRENT_LIST_DIR}
-            DEPENDS ${CMAKE_CURRENT_LIST_DIR}/pre_lex.py ${CMAKE_CURRENT_LIST_DIR}/${lex_file}.l
-            DEPFILE ${GENERATED_SRC}/${lex_file}.d
-            OUTPUT  ${GENERATED_SRC}/${lex_file}.l ${GENERATED_SRC}/${lex_file}.corr ${GENERATED_SRC}/${lex_file}.d
-        )
-    else()
-        add_custom_command(
-            COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_LIST_DIR}/pre_lex.py ${CMAKE_CURRENT_LIST_DIR}/${lex_file}.l  ${GENERATED_SRC}/${lex_file}.l ${GENERATED_SRC}/${lex_file}.corr ${GENERATED_SRC}/${lex_file}.d ${CMAKE_CURRENT_LIST_DIR}
-            DEPENDS ${CMAKE_CURRENT_LIST_DIR}/pre_lex.py ${CMAKE_CURRENT_LIST_DIR}/${lex_file}.l ${LEX_INC_FILES}
-            OUTPUT  ${GENERATED_SRC}/${lex_file}.l ${GENERATED_SRC}/${lex_file}.corr ${GENERATED_SRC}/${lex_file}.d
-        )
+        set(_depfile_args DEPFILE ${GENERATED_SRC}/${lex_file}.d)
     endif()
-    set_source_files_properties(${GENERATED_SRC}/${lex_file}.l PROPERTIES GENERATED 1)
-    set_source_files_properties(${GENERATED_SRC}/${lex_file}.corr PROPERTIES GENERATED 1)
-    set_source_files_properties(${GENERATED_SRC}/${lex_file}.d PROPERTIES GENERATED 1)
+    add_custom_command(
+        OUTPUT ${_generated_files}
+        COMMAND ${PYTHON_EXECUTABLE}
+            ${CMAKE_CURRENT_LIST_DIR}/pre_lex.py
+            ${CMAKE_CURRENT_LIST_DIR}/${lex_file}.l
+            ${_generated_files}
+            ${CMAKE_CURRENT_LIST_DIR}
+        DEPENDS
+            ${CMAKE_CURRENT_LIST_DIR}/pre_lex.py
+            ${CMAKE_CURRENT_LIST_DIR}/${lex_file}.l
+        ${_depfile_args}
+    )
+    set_source_files_properties(${_generated_files} PROPERTIES GENERATED 1)
```

Comment:
I left this is because it was being done before, but it _shouldn't_ be necessary. The `OUTPUT` files of `add_custom_command` will [automatically be marked `GENERATED`](https://cmake.org/cmake/help/latest/command/add_custom_command.html#generating-files:~:text=Each%20output%20file%20will%20be%20marked%20with%20the%20GENERATED%20source%20file%20property%20automatically.) by CMake itself. That's been documented behavior since at least CMake 3.0, if not earlier.

The only reason I can think of for re-applying the `GENERATED` property manually would be to ensure that it's visible in another scope. (And that's a relatively-recent CMake 3.18+ feature.) Like, this would ensure that the files are seen as `GENERATED` from the root build directory:
```cmake
if (CMAKE_VERSION VERSION_GREATER "3.18")
  set_source_files_properties(${_generated_files} 
    DIRECTORY ${CMAKE_BINARY_DIR}
    PROPERTIES GENERATED 1)
endif()
```

...But since CMake 3.20, [`GENERATED` is **always** globally-visible](https://cmake.org/cmake/help/latest/prop_sf/GENERATED.html#prop_sf:GENERATED), so even that's not really necessary.